backend/src/routes/file.ts

```diff
@@ -302,6 +351,42 @@ export class FileRoutes {
     const isModel = 'model' in req.query
     return dateforsize(isModel ? this.modelFileRepo : this.fileRepo, isModel ? 'model_file' : 'regular_file', req, res, next)
   }
+
+  private async deleteFileAndVisualizations(fileRepo: Repository<RegularFile|ModelFile>,
+    visualizationRepo: Repository<Visualization|ModelVisualization>,
+    uuid: string) {
+    await visualizationRepo.createQueryBuilder()
+      .delete()
+      .where({ sourceFile: uuid })
+      .execute()
+    await fileRepo.createQueryBuilder()
+      .delete()
+      .where({ uuid: uuid })
+      .execute()
+    await this.searchFileRepo.createQueryBuilder()
+      .delete()
+      .where({ uuid: uuid })
+      .execute()
+  }
+
+  private async getHigherLevelProducts(product: Product):Promise<string[]> {
+    // Returns Cloudnet products that are of higher level than the given product.
+    let uniqueLevels = await this.productRepo
+      .createQueryBuilder()
+      .select('DISTINCT level')
+      .orderBy('level')
+      .getRawMany()
+    uniqueLevels = uniqueLevels.map(level => level.level)
+    const index = uniqueLevels.indexOf(product.level)
+    const levels =  uniqueLevels.slice(index + 1)
+    let products = await this.productRepo
+      .createQueryBuilder()
+      .where({level: In(levels)})
```

Comment:
Tarvitaanko "where in" ja erillistä kyselyä? Eikö "where level > product.level" teksi saman asian?

Reply:
Hmm, pystyykö tekemään noin, kun meillä on sublevelit? 1b, 1c.. ? Voin toki kokeilla...

Reply:
Ai niin olikin. Jos nykyinen ratkaisi toimii, niin ei sitä tarvitse muuttaa.